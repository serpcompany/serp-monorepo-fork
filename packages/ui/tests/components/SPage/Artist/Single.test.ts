import { mockNuxtImport } from '@nuxt/test-utils/runtime';
import { describe, expect, it } from 'vitest';
import SPageArtistSingle from '../../../../components/SPage/Artist/Single.vue';
import ComponentRender from '../../../componentRender';
import '../../../mockUseUserSession';

mockNuxtImport('useSeoMeta', () => () => {});

let config_: Record<string, unknown> = {
  app: { baseURL: '/' },
  public: {
    useAuth: false
  }
};

let artistData_: unknown = {};
let upvotesData_: unknown = { upvotes: [] };

mockNuxtImport('useRuntimeConfig', () => () => config_);
mockNuxtImport('useArtist', () => () => artistData_);
mockNuxtImport('useFetchWithCache', () => () => upvotesData_);

const scenarios: [
  string,
  { config: Record<string, unknown>; artist: unknown; upvotes: string[] }
][] = [
  [
    'complete artist with upvotes and auth enabled',
    {
      config: { app: { baseURL: '/' }, public: { useAuth: true } },
      artist: {
        name: 'Test Artist',
        slug: 'test-artist',
        overview: '<p>Overview content for Test Artist.</p>',
        genres: ['Rock', 'Pop'],
        tags: ['Live', 'Studio'],
        beginArea: 'Test City',
        beginDate: '2000',
        endDate: '2020',
        releaseGroups: [
          {
            slug: 'album-1',
            cover_art_urls: { '500': '/album-1.jpg' },
            name: 'Album One',
            recordings: [
              {
                slug: 'song-1',
                has_lyrics: true,
                name: 'Song One',
                position: 1
              },
              {
                slug: 'song-2',
                has_lyrics: false,
                name: 'Song Two',
                position: 2
              }
            ]
          }
        ]
      },
      upvotes: ['user@example.com']
    }
  ],
  [
    'minimal artist without upvotes and auth disabled',
    {
      config: { app: { baseURL: '/' }, public: { useAuth: false } },
      artist: {
        name: 'Minimal Artist',
        slug: 'minimal-artist',
        overview: '<p>Minimal overview content.</p>',
        genres: [],
        tags: [],
        beginArea: '',
        beginDate: '',
        endDate: '',
        releaseGroups: []
      },
      upvotes: []
    }
  ]
];

describe('SPageArtistSingle Snapshot', () => {
  it.each(scenarios)(
    'renders %s correctly',
    async (desc: string, { config, artist, upvotes }) => {
      config_ = config;
      artistData_ = artist;
      upvotesData_ = { upvotes };

      const html = await ComponentRender(
        `SPageArtistSingle - ${desc}`,
        {},
        SPageArtistSingle
      );
      expect(html).toMatchSnapshot();
    }
  );
});
