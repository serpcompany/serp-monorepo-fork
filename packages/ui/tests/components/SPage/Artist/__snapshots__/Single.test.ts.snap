// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SPageArtistSingle Snapshot > renders complete artist with upvotes and auth enabled correctly 1`] = `
"<div>
  <header class="bg-background sticky top-0 z-40 border-b border-gray-200 transition-all duration-300 dark:border-gray-800 bg-background sticky top-0 z-10 transition-all duration-300">
    <div class="w-full max-w-(--ui-container) mx-auto px-4 sm:px-6 lg:px-8">
      <div class="flex flex-col items-center justify-between gap-4 py-3 sm:flex-row">
        <div class="flex min-w-0 flex-grow items-center gap-3">
          <!--v-if-->
          <div class="min-w-0 flex-grow">
            <h1 class="truncate text-xl font-bold text-gray-900 transition-all duration-300 sm:text-2xl dark:text-white">Test Artist</h1>
            <div class="truncate text-sm text-gray-500 dark:text-gray-400"></div>
          </div>
        </div>
        <div class="flex w-full flex-shrink-0 items-center justify-end gap-3 sm:w-auto">
          <div module="music_artists">
            <div class="relative inline-flex -space-x-px"><button type="button" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] text-sm gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted p-1.5"><span class="iconify i-lucide:arrow-big-up shrink-0 size-5" aria-hidden="true"></span>
                <!--v-if-->
                <!--v-if-->
              </button><span class="font-medium inline-flex items-center not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] text-xs gap-1 rounded-md ring ring-inset ring-accented text-default bg-default p-1 w-8 justify-center"><!--v-if--><span class="truncate">NaN</span>
              <!--v-if--></span><button type="button" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] text-sm gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted p-1.5"><span class="iconify i-lucide:arrow-big-down shrink-0 size-5" aria-hidden="true"></span>
                <!--v-if-->
                <!--v-if-->
              </button>
            </div>
          </div><a href="https://serp.ly/@daftfm/amazon/music/unlimited" rel="noopener noreferrer" target="_blank" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-2.5 py-1.5 text-xs gap-1.5 w-full sm:w-auto"><span class="iconify i-heroicons:arrow-top-right-on-square shrink-0 size-4" aria-hidden="true"></span><span class="truncate">Listen Free</span>
            <!--v-if-->
          </a>
        </div>
      </div>
      <nav class="-mx-4 flex overflow-x-auto border-t border-gray-200 px-4 sm:-mx-6 sm:px-6 dark:border-gray-800"><a href="#overview" class="font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-2.5 py-1.5 text-xs gap-1.5 flex-shrink-0 rounded-none whitespace-nowrap">
          <!--v-if--><span class="truncate">Overview</span>
          <!--v-if-->
        </a><a href="#albums" class="font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-2.5 py-1.5 text-xs gap-1.5 flex-shrink-0 rounded-none whitespace-nowrap">
          <!--v-if--><span class="truncate">Albums</span>
          <!--v-if-->
        </a><a href="#songs" class="font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-2.5 py-1.5 text-xs gap-1.5 flex-shrink-0 rounded-none whitespace-nowrap">
          <!--v-if--><span class="truncate">Songs</span>
          <!--v-if-->
        </a></nav>
    </div>
  </header><!-- Main content wrapper -->
  <div class="mx-auto max-w-7xl p-4 md:p-6 lg:p-8">
    <!-- Breadcrumbs -->
    <nav aria-label="breadcrumb" class="relative min-w-0 mb-6">
      <ol class="flex items-center gap-1.5"></ol>
    </nav><!-- Grid Layout -->
    <div class="grid gap-6 lg:grid-cols-3">
      <!-- Main Content Column (70%) -->
      <div class="space-y-8 lg:col-span-2">
        <!-- Albums Section - Refactored -->
        <section id="albums" class="scroll-mt-24" data-adsense-content="artist-albums">
          <h2 class="mb-4 text-2xl font-semibold">Test Artist Albums </h2>
          <div class="space-y-6">
            <div class="rounded-lg bg-default ring ring-default divide-y divide-default">
              <div class="p-4 sm:px-6">
                <div class="flex items-center justify-between"><a href="/albums/album-1" class="text-primary text-lg font-medium hover:underline">Album One</a>
                  <!--v-if-->
                </div>
              </div>
              <div class="p-4 sm:p-6">
                <div class="flex flex-col gap-4 sm:flex-row">
                  <!-- Album Cover -->
                  <!--v-if-->
                  <!-- Tracklist -->
                  <div class="flex-grow sm:w-3/4">
                    <h4 class="mb-2 text-sm font-medium text-gray-500 dark:text-gray-400"></h4>
                    <ol class="list-inside list-decimal space-y-1 text-sm">
                      <li><a href="/songs/song-1" class="text-primary-600 dark:text-primary-400 hover:underline">Song One</a></li>
                      <li><span class="text-gray-700 dark:text-gray-300">Song Two</span></li>
                    </ol>
                  </div>
                </div>
              </div>
              <!--v-if-->
            </div>
            <!--v-if-->
          </div>
        </section><!-- Songs section placeholder if needed separately -->
        <section id="songs" class="scroll-mt-24">
          <!-- Can list individual songs here if desired, separate from albums -->
        </section>
      </div><!-- Sidebar Column (30%) -->
      <div class="space-y-6 lg:col-span-1">
        <!-- Artist Image -->
        <!-- Fallback placeholder -->
        <div class="flex aspect-square items-center justify-center rounded-md bg-gray-100 shadow-sm dark:bg-gray-800"><span class="iconify i-heroicons:user h-1/3 w-1/3 text-gray-400 dark:text-gray-600" aria-hidden="true"></span></div><!-- Overview Card (Added to Sidebar) -->
        <div class="rounded-lg bg-default ring ring-default divide-y divide-default" data-adsense-content="artist-overview">
          <div class="p-4 sm:px-6">
            <div class="font-medium">Overview</div>
          </div>
          <div class="p-4 sm:p-6">
            <div class="prose prose-sm dark:prose-invert max-w-none">
              <p>Overview content for Test Artist.</p>
            </div>
          </div>
          <!--v-if-->
        </div><!-- General Info Card -->
        <div class="rounded-lg bg-default ring ring-default divide-y divide-default">
          <div class="p-4 sm:px-6">
            <div class="font-medium">Artist Info</div>
          </div>
          <div class="p-4 sm:p-6">
            <div class="space-y-2 text-sm">
              <div class="mb-2 text-lg font-semibold text-gray-900 dark:text-white">Test Artist</div>
              <!--v-if-->
              <div><span class="font-medium text-gray-700 dark:text-gray-300">Active From: </span><span class="text-gray-800 dark:text-gray-200">2000</span></div>
              <div><span class="font-medium text-gray-700 dark:text-gray-300">Active Until: </span><span class="text-gray-800 dark:text-gray-200">2020</span></div>
              <!--v-if-->
              <div><span class="font-medium text-gray-700 dark:text-gray-300">Origin: </span><span class="text-gray-800 dark:text-gray-200">Test City</span></div>
              <!--v-if-->
            </div>
          </div>
          <!--v-if-->
        </div><!-- Stats Card -->
        <div class="rounded-lg bg-default ring ring-default divide-y divide-default">
          <div class="p-4 sm:px-6">
            <div class="font-medium">Stats</div>
          </div>
          <div class="p-4 sm:p-6">
            <div class="space-y-2 text-sm">
              <div><span class="font-medium text-gray-700 dark:text-gray-300">Genres: </span><span class="text-gray-800 dark:text-gray-200">Rock, Pop</span></div>
              <div><span class="font-medium text-gray-700 dark:text-gray-300">Tags: </span><span class="text-gray-800 dark:text-gray-200">Live, Studio</span></div>
            </div>
          </div>
          <!--v-if-->
        </div><!-- Links Card -->
        <div class="rounded-lg bg-default ring ring-default divide-y divide-default">
          <div class="p-4 sm:px-6">
            <div class="font-medium">Listen On</div>
          </div>
          <div class="p-4 sm:p-6">
            <div class="p-1"><a href="https://serp.ly/@daftfm/amazon/music/unlimited" rel="noopener noreferrer" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-3 py-2 text-sm gap-2 w-full justify-center text-primary focus:outline-none focus-visible:bg-primary/15 disabled:bg-primary/10 aria-disabled:bg-primary/10 bg-blue-50 hover:bg-blue-100 dark:bg-blue-900/20 dark:hover:bg-blue-900/30">
                <!--v-if--> Amazon Music
                <!--v-if-->
              </a></div>
          </div>
          <!--v-if-->
        </div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`SPageArtistSingle Snapshot > renders minimal artist without upvotes and auth disabled correctly 1`] = `
"<div>
  <header class="bg-background sticky top-0 z-40 border-b border-gray-200 transition-all duration-300 dark:border-gray-800 bg-background sticky top-0 z-10 transition-all duration-300">
    <div class="w-full max-w-(--ui-container) mx-auto px-4 sm:px-6 lg:px-8">
      <div class="flex flex-col items-center justify-between gap-4 py-3 sm:flex-row">
        <div class="flex min-w-0 flex-grow items-center gap-3">
          <!--v-if-->
          <div class="min-w-0 flex-grow">
            <h1 class="truncate text-xl font-bold text-gray-900 transition-all duration-300 sm:text-2xl dark:text-white">Minimal Artist</h1>
            <div class="truncate text-sm text-gray-500 dark:text-gray-400"></div>
          </div>
        </div>
        <div class="flex w-full flex-shrink-0 items-center justify-end gap-3 sm:w-auto"><a href="https://serp.ly/@daftfm/amazon/music/unlimited" rel="noopener noreferrer" target="_blank" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-2.5 py-1.5 text-xs gap-1.5 w-full sm:w-auto"><span class="iconify i-heroicons:arrow-top-right-on-square shrink-0 size-4" aria-hidden="true"></span><span class="truncate">Listen Free</span>
            <!--v-if-->
          </a></div>
      </div>
      <nav class="-mx-4 flex overflow-x-auto border-t border-gray-200 px-4 sm:-mx-6 sm:px-6 dark:border-gray-800"><a href="#overview" class="font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-2.5 py-1.5 text-xs gap-1.5 flex-shrink-0 rounded-none whitespace-nowrap">
          <!--v-if--><span class="truncate">Overview</span>
          <!--v-if-->
        </a><a href="#albums" class="font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-2.5 py-1.5 text-xs gap-1.5 flex-shrink-0 rounded-none whitespace-nowrap">
          <!--v-if--><span class="truncate">Albums</span>
          <!--v-if-->
        </a><a href="#songs" class="font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-2.5 py-1.5 text-xs gap-1.5 flex-shrink-0 rounded-none whitespace-nowrap">
          <!--v-if--><span class="truncate">Songs</span>
          <!--v-if-->
        </a></nav>
    </div>
  </header><!-- Main content wrapper -->
  <div class="mx-auto max-w-7xl p-4 md:p-6 lg:p-8">
    <!-- Breadcrumbs -->
    <nav aria-label="breadcrumb" class="relative min-w-0 mb-6">
      <ol class="flex items-center gap-1.5"></ol>
    </nav><!-- Grid Layout -->
    <div class="grid gap-6 lg:grid-cols-3">
      <!-- Main Content Column (70%) -->
      <div class="space-y-8 lg:col-span-2">
        <!-- Albums Section - Refactored -->
        <section id="albums" class="scroll-mt-24" data-adsense-content="artist-albums">
          <h2 class="mb-4 text-2xl font-semibold">Minimal Artist Albums </h2>
          <div class="space-y-6">
            <p class="text-gray-500 dark:text-gray-400"> No albums found for this artist. </p>
          </div>
        </section><!-- Songs section placeholder if needed separately -->
        <section id="songs" class="scroll-mt-24">
          <!-- Can list individual songs here if desired, separate from albums -->
        </section>
      </div><!-- Sidebar Column (30%) -->
      <div class="space-y-6 lg:col-span-1">
        <!-- Artist Image -->
        <!-- Fallback placeholder -->
        <div class="flex aspect-square items-center justify-center rounded-md bg-gray-100 shadow-sm dark:bg-gray-800"><span class="iconify i-heroicons:user h-1/3 w-1/3 text-gray-400 dark:text-gray-600" aria-hidden="true"></span></div><!-- Overview Card (Added to Sidebar) -->
        <div class="rounded-lg bg-default ring ring-default divide-y divide-default" data-adsense-content="artist-overview">
          <div class="p-4 sm:px-6">
            <div class="font-medium">Overview</div>
          </div>
          <div class="p-4 sm:p-6">
            <div class="prose prose-sm dark:prose-invert max-w-none">
              <p>Minimal overview content.</p>
            </div>
          </div>
          <!--v-if-->
        </div><!-- General Info Card -->
        <div class="rounded-lg bg-default ring ring-default divide-y divide-default">
          <div class="p-4 sm:px-6">
            <div class="font-medium">Artist Info</div>
          </div>
          <div class="p-4 sm:p-6">
            <div class="space-y-2 text-sm">
              <div class="mb-2 text-lg font-semibold text-gray-900 dark:text-white">Minimal Artist</div>
              <!--v-if-->
              <!--v-if-->
              <!--v-if-->
              <!--v-if-->
              <!--v-if-->
              <!--v-if-->
            </div>
          </div>
          <!--v-if-->
        </div><!-- Stats Card -->
        <!--v-if-->
        <!-- Links Card -->
        <div class="rounded-lg bg-default ring ring-default divide-y divide-default">
          <div class="p-4 sm:px-6">
            <div class="font-medium">Listen On</div>
          </div>
          <div class="p-4 sm:p-6">
            <div class="p-1"><a href="https://serp.ly/@daftfm/amazon/music/unlimited" rel="noopener noreferrer" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-3 py-2 text-sm gap-2 w-full justify-center text-primary focus:outline-none focus-visible:bg-primary/15 disabled:bg-primary/10 aria-disabled:bg-primary/10 bg-blue-50 hover:bg-blue-100 dark:bg-blue-900/20 dark:hover:bg-blue-900/30">
                <!--v-if--> Amazon Music
                <!--v-if-->
              </a></div>
          </div>
          <!--v-if-->
        </div>
      </div>
    </div>
  </div>
</div>"
`;
